packages/wonder-blocks-link/components/__tests__/link.test.js

```diff
@@ -5,83 +5,310 @@ import {MemoryRouter, Route, Switch} from "react-router-dom";
-            </MemoryRouter>,
-        );
-
-        // Act
-        const buttonWrapper = wrapper.find(`[data-test-id="link"]`).first();
-        buttonWrapper.simulate("click", {button: 0});
-
-        // Assert
-        expect(wrapper.find("#foo").exists()).toBe(false);
+    describe("server-side navigation", () => {
+        test("doesn't redirect if safeWithNav hasn't resolved yet when skipClientNav=true", () => {
+            // Arrange
+            jest.spyOn(window.location, "assign").mockImplementation(() => {});
+            const wrapper = mount(
+                <Link
+                    testId="link"
+                    href="/foo"
+                    safeWithNav={() => Promise.resolve()}
+                    skipClientNav={true}
+                >
+                    Click me!
+                </Link>,
+            );
+
+            // Act
+            const buttonWrapper = wrapper.find(`[data-test-id="link"]`).first();
+            buttonWrapper.simulate("click", {
+                button: 0,
+            });
+
+            // Assert
+            expect(window.location.assign).not.toHaveBeenCalled();
+        });
+
+        test("redirects after safeWithNav resolves when skipClientNav=true", async () => {
+            // Arrange
+            jest.spyOn(window.location, "assign").mockImplementation(() => {});
+            const wrapper = mount(
+                <Link
+                    testId="link"
+                    href="/foo"
+                    safeWithNav={() => Promise.resolve()}
+                    skipClientNav={true}
+                >
+                    Click me!
+                </Link>,
+            );
+
+            // Act
+            const buttonWrapper = wrapper.find(`[data-test-id="link"]`).first();
+            buttonWrapper.simulate("click", {
+                button: 0,
+            });
+            await wait(0);
+            wrapper.update();
+
+            // Assert
+            expect(window.location.assign).toHaveBeenCalledWith("/foo");
+        });
+
+        test("redirects after beforeNav and safeWithNav resolve when skipClientNav=true", async () => {
+            // Arrange
+            jest.spyOn(window.location, "assign").mockImplementation(() => {});
+            const wrapper = mount(
+                <Link
+                    testId="link"
+                    href="/foo"
+                    beforeNav={() => Promise.resolve()}
+                    safeWithNav={() => Promise.resolve()}
+                    skipClientNav={true}
+                >
+                    Click me!
+                </Link>,
+            );
+
+            // Act
+            const buttonWrapper = wrapper.find(`[data-test-id="link"]`).first();
+            buttonWrapper.simulate("click", {
+                button: 0,
+            });
+            await wait(0);
+            wrapper.update();
+
+            // Assert
+            expect(window.location.assign).toHaveBeenCalledWith("/foo");
+        });
```

Comment:
I'd like to see a test for `doesn't redirect before beforeNav resolves when skipClientNav=true` in addition to the ones that already exist.